syntax = "proto3";

message addClient{
  string client_name=1;
  string client_address_and_port=2;
}

message clientName {
  string client_name=1;
}

message getConnectionInfo{
  string client_name=1;
  string client_address_and_port=2;
}

message infoFromNameServer{
  int32 accepted=1;
}

service NameServer {
  rpc getClientInfo(clientName) returns (getConnectionInfo);
  rpc addToNameServer(addClient) returns (infoFromNameServer);
}

message Empty{

}